lib/weather_web/router.ex

```diff
@@ -2,48 +2,43 @@ defmodule WeatherWeb.Router do
   use WeatherWeb, :router
 
   # Aliases
-  alias WeatherGraph.Schema
-  alias WeatherWeb.{Plugs, DataController, UserSocket}
-
+  alias Weather.Authentication
+  alias WeatherGraph.{Plugs, Schema}
+  alias WeatherWeb.{DataController, UserSocket}
   # Pipelines
   pipeline :api do
-    plug(:accepts, ["json"])
+    plug :accepts, ["json"]
   end
 
   pipeline :graphql do
-    plug(:accepts, ["json"])
+    plug :accepts, ["json"]
 
-    plug(Plugs.Context)
+    plug Authentication.Pipeline
+    plug Plugs.Context
   end
 
   scope "/api" do
-    pipe_through(:api)
+    pipe_through :api
 
-    resources "/datas", DataController, only: [:create]
+    post "/stations/:station_id/datas", DataController, :create
   end
 
   scope "/graphql" do
-    pipe_through(:graphql)
-
-    forward(
-      "/",
-      Absinthe.Plug,
-      schema: Schema,
-      socket: UserSocket
-    )
+    pipe_through :graphql
+
+    forward "/", Absinthe.Plug,
+        schema: Schema,
+        socket: UserSocket
   end
 
   with :dev <- Mix.env() do
     scope "/graphiql" do
-      pipe_through(:graphql)
+      pipe_through :graphql
 
-      forward(
-        "/",
-        Absinthe.Plug.GraphiQL,
+      forward "/", Absinthe.Plug.GraphiQL,
         schema: Schema,
         socket: UserSocket,
-        interface: :advanced
-      )
+        interface: :playground
```

Comment:
Is playground even more useful than advanced?

Reply:
No but look better, check the sceenshot.